src/app/shared/services/local-storage.ts

```diff
@@ -0,0 +1,17 @@
+export const LocalStorage = {
+  set(key: string, value: any) {
+    window.localStorage.setItem(
```

Comment:
this may cause an issue in some cases, so direct window, DOM, document, localStorage is bad practice in Angular
try this solution 
![image](https://user-images.githubusercontent.com/47972692/206440544-5edb9c29-ea09-4926-a184-9b38757dbb79.png)
from here https://stackoverflow.com/questions/39085632/localstorage-is-not-defined-angular-universal

Reply:
This fix use code what 5 years deprecated code. "OpaqueToken" is deprecated. Additional to this fix based on deprecated project snippet, so some part of resolve can't be complete.
https://forum.ionicframework.com/t/module-node-modules-angular-core-index-has-no-exported-member-opaquetoken/117775/2

If I correctly remember we use Local Storage only to store information between several tabs and to load user what login before, do we have other way to store this data and load after page closed but without use Local Storage?